### PR TITLE
New build tags job

### PR DESCRIPTION
**Pull request by aduprat**
Only last commit: ea57d3f

Tests validated:
- unknown tag -> build fails
- tag with compilation error -> build fails
- tag is never merged on master (merging this tag will fail) -> build works
- tag on top of master -> build works

- standard branch build  -> build works
- branch with conflict when merging -> build fails

The docker images produced by the build of a tag are successfully published.

### CODE REVIEW

File: workflow-job [hunk 1]
```diff
@@ -448,7 +467,12 @@ flows["${sha1}"] = {
 
                 if (packagageOnGitlab()) {
                     wrap([$class: 'MaskPasswordsBuildWrapper', varPasswordPairs: [[password: "${gitlabToken}", var: 'gitlabToken']]]) {
-                        sh "curl -X POST -Ftoken=${gitlabToken} -F'ref=master' -F'variables[REPO_NAME]=oncommit' -F'variables[SOURCE_REPO]=https://github.com/linagora/james-project' https://ci.linagora.com/api/v4/projects/803/trigger/pipeline"
+                        if (isMaster()) {
+                            sh "curl -X POST -Ftoken=${gitlabToken} -F'ref=master' -F'variables[REPO_NAME]=oncommit' -F'variables[SOURCE_REPO]=https://github.com/linagora/james-project' https://ci.linagora.com/api/v4/projects/803/trigger/pipeline"
+                        }
+                        if (isTag()) {
+                            sh "curl -X POST -Ftoken=${gitlabToken} -F'ref=master' -F'variables[REPO_NAME]=oncommit' -F'variables[SOURCE_REPO]=https://github.com/linagora/james-project' -F'variables[TAG]=${tagName} https://ci.linagora.com/api/v4/projects/803/trigger/pipeline"
```
Comment (rouazana) [inline workflow-job]: this should not be on the oncommit REPO, but on the jamesxxx (where xxx is the openpaas version). Could be tricky to compute this...

File: workflow-job [hunk 2]
```diff
@@ -471,7 +495,22 @@ def merge() {
...
+    git url: authorParentURL, branch: "master"
+    sh "git fetch --tags origin"
+    sh "git checkout ${tagName} -b ${mergeBranch}"
+    jamesVersion = version()
+    def dockerfile = """\
+                        FROM debian:8.11
+                        
+                        ADD . /origin
+                        WORKDIR /origin
+                        CMD echo \"Volume built\"""".stripIndent()
```
Comment (rouazana) [inline workflow-job:505]: debug?

**Review by rouazana — COMMENTED** [review]
If you have a good way to fix the packaging issue, it would be great, else maybe we can postpone it to another ticket.

File: workflow-job (hunk 2 above)
Comment (aduprat) [reply workflow-job:505]: No, it's just the command inside the container without side effect.
It's a trick in order to build a volume

**Comment by aduprat** [comment]
> If you have a good way to fix the packaging issue, it would be great, else maybe we can postpone it to another ticket.

Yes, we will do that later

**Review by rouazana — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Pull request merged**